### PR TITLE
Add torchvision

### PR DESCRIPTION
**Pull request by sigseg5**

### CODE REVIEW

File: projects/vision/decode_jpeg_fuzz.cc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  fp.write((char *)(data + 1), size_i);
+  fp.close();
+
+  if (size <= 0) {
+    unlink(dir);
+    return 0;
+  }
+
+  auto input_data =
+      torch::from_file(dir, /*shared=*/false, /*size=*/size_i, torch::kU8);
```
Comment (anfedotoff) [inline projects/vision/decode_jpeg_fuzz.cc:36]: Is it possible to create tensor from memory?
https://github.com/pytorch/pytorch/blob/a7edf713608806f10e17eab90d0a5df727d9a16e/aten/src/ATen/native/TensorFactories.cpp#L1463

File: projects/vision/decode_jpeg_fuzz.cc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//###############################################################################
+
+#include "decode_jpeg.h"
+#include <unistd.h>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  if (size <= 2) { return 0; }
```
Comment (SweetVishnya) [inline projects/vision/decode_jpeg_fuzz.cc:21]: size < 2

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: projects/vision/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+RUN git clone https://github.com/pytorch/vision.git vision_fuzz/
+RUN cd /vision_fuzz && \
+    git checkout c3ec69482a5b9359306c58f0f5c587e1b3bb11cb
+
+# Patch torchvision CMakeLists to build static
+RUN sed -i 's/add_library(${PROJECT_NAME} SHARED ${ALL_SOURCES})/add_library(${PROJECT_NAME} STATIC ${ALL_SOURCES})/' /vision_fuzz/CMakeLists.txt && \
+    sed -i 's/target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIBRARIES} ${PNG_LIBRARY} ${JPEG_LIBRARIES})/add_definitions(-DJPEG_FOUND -DPNG_FOUND)/' /vision_fuzz/CMakeLists.txt
+
+# Install OpenCV
+RUN cd / && mkdir cv && cd cv/ && wget -O opencv.zip https://github.com/opencv/opencv/archive/4.x.zip && unzip opencv.zip && mkdir -p build && cd build && cmake -DBUILD_TESTS=OFF ../opencv-4.x && cmake --build . -j$(nproc) && make install && cd /
```
[the remark this answers is not in the document]
Comment (sigseg5) [reply projects/vision/Dockerfile]: Done

File: projects/vision/build_corpus.sh [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+#
+################################################################################
+
+#!/bin/bash -eu
+
+# Build and link jpeg tensor saver
+
+cd /vision_sydr/build 
+
+clang++ -DAT_PER_OPERATOR_HEADERS -DCPUINFO_SUPPORTED_PLATFORM=1 -DFMT_HEADER_ONLY=1 -DFXDIV_USE_INLINE_ASSEMBLY=0 -DHAVE_MALLOC_USABLE_SIZE=1 -DHAVE_MMAP=1 -DHAVE_SHM_OPEN=1 -DHAVE_SHM_UNLINK=1 -DMINIZ_DISABLE_ZIP_READER_CRC32_CHECKS -DNNP_CONVOLUTION_ONLY=0 -DNNP_INFERENCE_ONLY=0 -DONNXIFI_ENABLE_EXT=1 -DONNX_ML=1 -DONNX_NAMESPACE=onnx_torch -DUSE_EXTERNAL_MZCRC -D_FILE_OFFSET_BITS=64 -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -DTH_HAVE_THREAD -g -O2 -I/libjpeg-turbo-2.1.3-sydr/ -I/pytorch_sydr/torch/include /save_jpeg2tensor.cc -I/pytorch_sydr/torch/csrc/api/include -I/vision_sydr/torchvision/csrc/io/image -I/vision_sydr/torchvision/csrc/io/image/cpu -c -o ./save_jpeg.o
```
[the remark this answers is not in the document]
Comment (sigseg5) [reply projects/vision/build_corpus.sh]: Done all multilines

File: projects/vision/decode_jpeg_fuzz.cc (hunk 2 above)
Comment (sigseg5) [reply projects/vision/decode_jpeg_fuzz.cc]: Done

File: projects/vision/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+     save_jpeg2tensor.cc \
+     save_png2tensor.cc \
+     *.toml \
+     /
+
+# Apply PyTorch fixes.
+RUN cd /pytorch_fuzz && \
+    git apply /stoull.patch
+RUN rm /pytorch_fuzz/third_party/miniz-2.0.8/miniz.*
+RUN cp miniz.* /pytorch_fuzz/third_party/miniz-2.0.8/.
```
[the remark this answers is not in the document]
Comment (sigseg5) [reply projects/vision/Dockerfile]: Done

File: projects/vision/README.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    # sydr-fuzz -c <target_name>.toml security
+
+## Crash analysis with Casr
+
+    # sydr-fuzz -c <target_name>.toml casr
+
+## Applied patches for PyTorch
+
+* miniz.* – Updated miniz version to fix segmentation fault.
+* stoull.patch – Catch stoull exception to allow the fuzzer go deeper.
```
[the remark this answers is not in the document]
Comment (sigseg5) [reply projects/vision/README.md]: Done

**Pull request merged**